ebuild-writing/functions/src_compile/building/text.xml

```diff
@@ -60,7 +60,7 @@ src_prepare() {
 	# We have a weird Makefile to work with which ignores our
 	# compiler preferences. yay!
 	# Note the single quotes.
-	sed -i -e 's:cc -O2:$(CC) $(CFLAGS) $(LDFLAGS):' Makefile \
+	sed -i -e 's;cc -O2;$(CC) $(CFLAGS) $(LDFLAGS);' Makefile \
```

Comment:
Why this change? We should stick to the common separator characters (like `/` or `:`). Unusual ones like `;` only make the reader wonder if there's a reason for it.

Reply:
Did you read the commit message? :)

```
Author: Sam James <sam@gentoo.org>
Date:   Sat Mar 20 09:55:09 2021 +0000

    ebuild-writing/functions/src_compile/building: use ; in CFLAGS sed

    A colon is a valid character in CFLAGS, so we shouldn't use it
    as a sed delimiter.

    Bug: https://bugs.gentoo.org/685160
    Signed-off-by: Sam James <sam@gentoo.org>
```

Reply:
I don't see any colon in `$(CC) $(CFLAGS) $(LDFLAGS)` (and neither in `cc -O2`).

Maybe I need new glasses? :smile:

Reply:
Ah, I think I understand now.
Note the single quotes, `$(CFLAGS)` is a literal string here.

Reply:
Also, if it wasn't a literal string, then `;` would only marginally be better than `:`. CFLAGS originates from user configuration, so one cannot assume anything about it, and it may contain _any_ character. Expanding the variable in a sed replacement is simply asking for trouble and should be avoided.

Reply:
Ahhh, I missed it was literal! It's still extremely common to end up sedding CFLAGS but I take the point that it's a rabbit hole.

I'd still say that using something other than a comma or a colon is ideal for where people don't want to fix the build system though. The example in the devmanual should probably be 'less wrong', but maybe it'd be useful to put a note about this as well somewhere?

Reply:
Or maybe just a warning in the sed section about choosing your delimiter?

Reply:
We say something about it [here](https://devmanual.gentoo.org/tools-reference/sed/index.html#simple-text-substitution-using-sed).

Reply:
I've added some more information there.